pkg/agents/team5/social.go

```diff
@@ -72,6 +72,78 @@ func (a *CustomAgent5) updateFavour() {
 	}
 }
 
+func (a *CustomAgent5) calculateFavourAbove(aboveID uuid.UUID) int {
+	sens := a.sensitivity
+	foodIntent := float64(a.attemptFood)
+	hpSelf := float64(a.HP())
+	// I would still add a check here to make sure it isn't -1.
+	foodPlat := float64(a.CurrPlatFood())
+
+	maxHP := float64(a.HealthInfo().MaxHP)
+
+	hpAbove := float64(a.socialMemory[aboveID].agentHP)
+	foodAbove := float64(a.socialMemory[aboveID].foodTaken)
+	foodExp := float64(a.avgPlatFood)
+
+	F_expectedComponent := math.Exp(foodPlat/(3.0*foodExp+15.0)) * (foodPlat - foodExp) / (50 * sens)
+	H_aboveScoreComponent := -2 * sens * hpAbove * math.Pow(foodAbove, 2) / math.Pow(maxHP, 3)
+	H_ourScoreComponent := 2 * sens * math.Pow(foodIntent, 1.3) * math.Pow(hpSelf, 1.7) / math.Pow(maxHP, 3)
+
+	change := F_expectedComponent + H_aboveScoreComponent + H_ourScoreComponent
+	return int(math.Round(change))
+}
+
+func (a *CustomAgent5) calculateExpectedFood() float64 {
+	// average of food taken by agents around our agent, including our agent
+	foodTotal := a.lastMeal
+	for _, uuid := range a.surroundingAgents {
+		foodTotal += a.socialMemory[uuid].foodTaken
+	}
+	foodExp := float64(foodTotal) / float64(len(a.surroundingAgents)+1)
+	if foodExp == 0 {
+		foodExp = float64(a.attemptFood) // to avoid favour shooting extremely high
+	}
+	return foodExp
+}
+
+func (a *CustomAgent5) calculateFavourBelow(belowID uuid.UUID) (int, food.FoodType) {
+	sens := a.sensitivity
+	foodIntent := float64(a.attemptFood)
+	hpSelf := float64(a.HP())
+
+	maxHP := float64(a.HealthInfo().MaxHP)
+
+	hpBelow := float64(a.socialMemory[belowID].agentHP)
+	foodBelow := float64(a.socialMemory[belowID].foodTaken)
```

Comment:
We call these functions when the platform is on our floor, so the agent below would not have eaten that day. Is this acceptable that this will use the food they ate in the previous day?

Reply:
its acceptable cause the favour change due to their hp and foodtaken is small and will only lag behind by 1 day

Reply:
but if its possible to do like you say then its probably better